app/src/main/java/org/mozilla/fenix/utils/Settings.kt

```diff
@@ -107,9 +108,10 @@ class Settings(private val appContext: Context) : PreferencesHolder {
     override val preferences: SharedPreferences =
         appContext.getSharedPreferences(FENIX_PREFERENCES, MODE_PRIVATE)
 
-    var showTopFrecentSites by booleanPreference(
+    var showTopFrecentSites by lazyFeatureFlagPreference(
         appContext.getPreferenceKey(R.string.pref_key_enable_top_frecent_sites),
-        default = true
+        featureFlag = true,
+        default = { appContext.components.analytics.features.homeScreen.isTopSitesActive() }
```

Comment:
For top sites, I didn't have to wired up the functionality on the home screen as it was already implemented previously see references [one](https://github.com/mozilla-mobile/fenix/blob/782aea2d5bdc74eee718529ff53218949b4e2389/app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt#L271-L273) and [two](https://github.com/mozilla-mobile/fenix/blob/782aea2d5bdc74eee718529ff53218949b4e2389/app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt#L394).